Ext4Pkg/Ext4Dxe/BlockMap.c

```diff
@@ -228,18 +228,18 @@ Ext4GetBlocks (
 
   Inode = File->Inode;
 
-  BlockPathLength = Ext4GetBlockPath (Partition, LogicalBlock, BlockPath);
+  BlockPathLength = Ext4GetBlockPath (Partition, (UINT32) LogicalBlock, BlockPath);

```

Comment:
Iirc there is a check somewhere to verify they're in bounds for 32-bit. Maybe the casting should happen a lot further up the call stack and all functions take UINT32.

Reply:
Ext4GetExtent calls Ext4GetBlocks conditionally checking EXT4_EXTENTS_FL.
https://github.com/acidanthera/audk/blob/668cb050e3cdf1eca17200e342126f40a7ec6aa2/Ext4Pkg/Ext4Dxe/Extents.c#L260 -> We can cast to UINT32 there, and make LogicalBlock 32-bit in blockmap

Reply:
Made change in https://github.com/acidanthera/audk/pull/11/commits/70ac4ec242965ae4e1e35b1fb46b8aac7c7671a2